srv/components/system/network/prepare.sls

```diff
@@ -41,28 +41,30 @@ Stop and disable NetworkManager service:
 
 # Disabling NetworkManager doesn't kill dhclient process.
 # If not killed explicitly, it causes network restart to fail: COSTOR-439
-Kill dhclient:
-  cmd.run:
-    - name: pkill -SIGTERM dhclient
-    - onlyif: pgrep dhclient
-    - requires:
-      - service: Stop and disable NetworkManager service
+# Kill dhclient:
+#   cmd.run:
+#     # - name: pkill -SIGTERM dhclient
+#     - name: dhclient -x
+#     - onlyif: pgrep dhclient
+#     - requires:
+#       - service: Stop and disable NetworkManager service
 
-Start dhclient:
-  cmd.run:
-    # - name: dhclient {{ pillar['cluster'][grains['id']]['network']['mgmt_nw']['iface'][0] }}
-    - name: dhclient -H {{ grains["host"] }}
-    - unless: pgrep dhclient
-    - listen:
-      - Kill dhclient
+{#% if not pillar['cluster'][grains['id']]['network']['mgmt_nw']['public_ip_addr'] %#}
+# Start dhclient:
+#   cmd.run:
+#     - name: dhclient -1 -q -H $(hostname -s) {{ pillar['cluster'][grains['id']]['network']['mgmt_nw']['iface'][0] }}
+#     - unless: pgrep dhclient
+#     - listen:
+#       - Kill dhclient
+{#% endif %#}
 
-{% else  %}
+{#% else  %#}
 
-network_manager_is_not_running:
-  test.nop: []
+# network_manager_is_not_running:
+#   test.nop: []
 
-{% endif %}
+{#% endif %#}
 
-# Dummy placeholder for network.prepare:
-#   test.show_notification:
-#     - text: "A yaml file with comments results in minion non-zero exit"
+Dummy placeholder for network.prepare:
```

Comment:
Instead we can skip running prepare state from init.

Reply:
This is necessary as there are multiple calls internally and there's possibility of direct call somewhere.
But, I've commented it in init.sls as well